hta/common/trace_file.py

```diff
@@ -25,53 +26,49 @@ def create_rank_to_trace_dict(trace_path: str) -> Tuple[bool, Dict]:
-        for fn in os.listdir(trace_path)
-        if fn.endswith(".gz") or fn.endswith(".json")
+        fn for fn in os.listdir(trace_dir) if fn.endswith(".gz") or fn.endswith(".json")
     ]
     if len(file_list) == 0:
-        logger.warning(f"No trace file is found in {trace_path}")
+        logger.warning(f"No trace file is found in {trace_dir}")
         return False, {}
 
     rank_to_trace_dict: Dict[int, str] = {}
     for file in file_list:
-        filename = os.path.join(trace_path, file)
-        data = None
-        if file.endswith("gz"):
-            gzip_file_handle = gzip.open(filename, "rb")
-            data = json.loads(gzip_file_handle.read())
-        if file.endswith("json"):
-            json_file_handle = open(filename, "r")
-            data = json.loads(json_file_handle.read())
-
-        distributed_info = data.get("distributedInfo", None)
-        if distributed_info is None:
-            logger.error(
-                "If the trace file does not have the rank specified in it, then add the following snippet "
-                'key to the json files to use HTA; "distributedInfo": {"rank": 0}. If there are multiple '
-                "traces files, then each file should have a unique rank value."
-            )
-            sys.exit()
-
-        rank = distributed_info.get("rank", None)
-        if rank is None:
-            logger.error(
-                "If the trace file does not have the rank specified in it, then add the following snippet "
-                'key to the json files to use HTA; "distributedInfo": {"rank": 0}. If there are multiple '
-                "traces files, then each file should have a unique rank value."
-            )
-            sys.exit()
-
-        assert isinstance(rank, int), "Rank is expected to be an integer"
-        assert rank >= 0, "Rank must be a non-negative integer"
-        rank_to_trace_dict[rank] = filename
-    if len(rank_to_trace_dict) == 0:
-        logger.warning("Rank to trace dict has size zero.")
+        file_path = os.path.join(trace_dir, file)
+
+        with gzip.open(file_path, "rb") if file.endswith("gz") else open(
+            file_path, "r"
+        ) as f:
+            for line in f:
+                data = line.decode() if isinstance(line, bytes) else line
+                if "rank" in data:
+                    break
+
+            # match is like "rank": 6,
+            match = re.search(r'"rank": \d+', data)
+            if match:
+                rank = match.group().split(": ")[1]
+                rank_to_trace_dict[int(rank)] = file_path
+            else:
+                logger.error(
+                    "If the trace file does not have the rank specified in it, then add the following snippet "
+                    'key to the json files to use HTA; "distributedInfo": {"rank": 0}. If there are multiple '
+                    "traces files, then each file should have a unique rank value."
+                )
+                sys.exit()
```

Comment:
You may remove the sys.exit() because under certain scenarios not all files within trace_dir are trace files.